bot/exts/moderation/modpings.py

```diff
@@ -116,16 +121,26 @@ async def add_role_schedule(self, mod: Member, work_time: int, schedule_start: d
             log.trace(f"Skipping adding moderator role to mod with ID {mod.id} - found in pings off cache.")
         else:
             log.trace(f"Applying moderator role to mod with ID {mod.id}")
-            await mod.add_roles(self.moderators_role, reason="Moderator scheduled time started!")
+            await self.reapply_role(mod, reason="Moderator scheduled time started!")
 
         log.trace(f"Sleeping for {work_time} seconds, worktime for mod with ID {mod.id}")
         await asyncio.sleep(work_time)
         await self.remove_role_schedule(mod, work_time, schedule_start)
 
-    async def reapply_role(self, mod: Member) -> None:
+    async def reapply_role(self, mod: Member, *, reason: str = None) -> None:
         """Reapply the moderator's role to the given moderator."""
+        await self.fetch_guild_task
+        if self.guild.get_role(Roles.mod_team) not in mod.roles:
+            log.info(
```

Comment:
I think this should be upgraded to a `log.warning`, since it means we likely need to `int e` remove them from the modpings.

Reply:
https://github.com/python-discord/bot/pull/2008/commits/1266c5321253da1817d1550efb8008000901a64d